### PR TITLE
Show cursor when using inverted mouse dragging

### PR DESCRIPTION
**Pull request by beam41**
I add this because many modern management game has grab and drag control, which in my opinion has better user experience than hiding cursor.
 
openrct2

image

this option is disable when user does not enable invert right mouse dragging (because it look weird). So I add little indent in option menu.

The phrase I use in menu and tooltip might need changes, but I can't come up with some things better.

### CODE REVIEW

**Comment by jensj12** [comment]
This change is probably too small to warrant an extra entry in the options. That said, I think this change makes sense and could be merged without it being an option. It should have been like that since touchscreen-like dragging was made available.
(you might still want to wait for one of the devs to confirm before making changes, and I don't use inverted scrolling anyways so don't count my opinion)

**Comment by Gymnasiast** [comment]
I agree, make it unconditional.

**Comment by beam41** [comment]
Might need to update tooltip then.

Would be nice if we have interface that we can throw any setting in like openttd.

**Review by rik-smeets — CHANGES_REQUESTED** [review]
@beam41 I agree with @Gymnasiast , just show the cursor when dragging and drop the options toggle for it entirely. Could you change that please?

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -2985,7 +2985,7 @@ STR_5822    :Cycle between day and night.{NEWLINE}Full cycle takes one in-game m
 STR_5823    :Display banners in uppercase (RCT1 behaviour)
 STR_5824    :Disables lightning effect{NEWLINE}during a thunderstorm
 STR_5825    :Keep the mouse cursor in the window
-STR_5826    :Invert right mouse dragging of the viewport
+STR_5826    :Invert right mouse dragging of the viewport. Enable this will not hide cursor on drag.
```
Comment (rik-smeets) [inline data/language/en-GB.txt:2988]: Please revert this change, it isn't necessary anymore

File: src/openrct2-ui/input/MouseInput.cpp [hunk 2]
```diff
@@ -526,7 +526,10 @@ static void InputViewportDragBegin(rct_window& w)
     _ticksSinceDragStart = 0;
     auto cursorPosition = context_get_cursor_position();
     gInputDragLast = cursorPosition;
-    context_hide_cursor();
+    if (!gConfigGeneral.invert_viewport_drag)
```
Comment (rik-smeets) [inline src/openrct2-ui/input/MouseInput.cpp:529]: I think you forget to remove this if statement, same for the other ones below. Just remove the `context_hide_cursor()` call in this function and that should be it.

**Review by rik-smeets — CHANGES_REQUESTED** [review]

File: src/openrct2-ui/input/MouseInput.cpp (hunk 2 above)
Comment (jensj12) [reply src/openrct2-ui/input/MouseInput.cpp:529]: Doesn't removing the call entirely also remove it for right-mouse panning (or non-inverted dragging)? I'd rather keep it as-is when inverted right-mouse dragging is disabled.

Comment (rik-smeets) [reply src/openrct2-ui/input/MouseInput.cpp:529]: You're right, I personally never drag the screen around (only use the keyboard) so I missed that nuance. Thanks!

In that case, @beam41, please fix the compile errors by replacing `invert_viewport_drag` with `InvertViewportDrag`.

**Review by rik-smeets — APPROVED** [review]
